### ISSUE TITLE
Cryptography related dependencies are not represented correctly 

### ISSUE THREAD

**Issue by n1ckl0sk0rtge** [cbomkit/sonar-cryptography#66]
If cryptographic values are interdependent, the “internal” cryptographic references should be used to define these relationships. 

### Example:
if a `key` dependence on an `algorithm` use the `cryptoRef` field in the `relatedCryptoMaterialsProperties` to reference the algorithm.

**Comment by n1ckl0sk0rtge** [comment]
At the moment, this topic is on hold. In the current version of CBOM, there is no option to reference algorithms from other algorithms. This is necessary for many scenarios (signature algorithm uses digest algorithm). As this reference cannot be expressed via the algorithm properties and for reasons of conciseness, we present all dependencies in the Dependencies section.

**Comment by sachin9058** [comment]
Hi @n1ckl0sk0rtge I would like to work on this issue..

**Comment by sachin9058** [comment]
Hi @n1ckl0sk0rtge, I spent some time looking into this issue and the related CBOM representation constraints.

From my understanding, the current challenge is that algorithm-to-algorithm relationships (for example, a signature algorithm depending on a digest algorithm like RSA-PSS → SHA256) cannot currently be expressed directly through algorithm properties, which is why dependencies are currently emitted only in the Dependencies section.

I’d like to explore possible approaches for representing these relationships more explicitly while still remaining aligned with the current CBOM model. One idea could be introducing internal references between cryptographic components during translation/output generation, especially for composed algorithms and signature schemes.

I’m going through the mapper/output flow to better understand how dependencies are currently constructed and where these relationships could be attached safely.

Happy to discuss possible directions before starting implementation work.

**Comment by sachin9058** [comment]
Hi @n1ckl0sk0rtge, I investigated this issue further.

I traced the output flow and found that `cryptoRefArray` support currently exists for `ProtocolProperties` (used in protocol output generation), but `AlgorithmProperties` does not expose equivalent support.

I also checked the CycloneDX model directly (`AlgorithmProperties`) and could not find `setCryptoRefArray(...)`, which seems consistent with the earlier note that algorithm-to-algorithm references cannot currently be expressed through algorithm properties.

At the moment it looks like relationships such as `RSASSA-PSS -> SHA256 / MGF1 / RSA` can only be represented through dependencies unless the underlying model/schema support changes.

**Comment by n1ckl0sk0rtge** [comment]
Unblocking this.

The reason it was on hold was that CBOM had no way to reference one crypto asset from another. CycloneDX 1.7 fixes that. It adds `relatedCryptographicAssets`, a generic `{type, ref}` relationship list, and deprecates the ad-hoc ref fields we use today (`signatureAlgorithmRef`, `subjectPublicKeyRef`, `relatedCryptoMaterialProperties.algorithmRef`, `protocolProperties.cryptoRefArray`).

So the case from the first comment — a signature algorithm that uses a digest algorithm — can now be expressed directly, instead of only through the Dependencies section.

This work depends on the 1.7 upgrade in #505. Order:

1. #505 switches the output to 1.7 and moves the deprecated ref fields to `relatedCryptographicAssets`.
2. This issue then adds the real asset-to-asset relations we could not express before.

@sachin9058 — your finding was right: `AlgorithmProperties` has no `setCryptoRefArray(...)` in the 1.6 model. That is exactly what 1.7 changes. If you are still interested, this is now doable, but it should start after #505 is in.
